indra/newview/scripts/lua/test_flycam.lua

```diff
@@ -0,0 +1,38 @@
+-- Make camera fly around the subject avatar for a few seconds.
+
+local LLAgent = require 'LLAgent'
+local startup = require 'startup'
+local timers = require 'timers'
+
+local initial = LLAgent.getRegionPosition()
+local height = 2.0 -- meters
+local radius = 4.0 -- meters
+local speed = 1.0 -- meters/second along circle
+local start = os.clock()
+local stop  = os.clock() + 30 -- seconds
+
+local function cameraPos(t)
+    local radians = speed * t
+    return {
+        initial[1] + radius * math.cos(radians),
+        initial[2] + radius * math.sin(radians),
+        initial[3] + height
+    }
+end
+
+local function moveCamera()
+    if os.clock() < stop then
+        -- usual case
+        LLAgent.setCamera{ camera_pos=cameraPos(os.clock() - start), camera_locked=true }
+        return nil
+    else
+        -- last time
+        LLAgent.removeCamParams()
+        LLAgent.setFollowCamActive(false)
+        return true
+    end
+end
```

Comment:
I think I still don't understand follow cam behavior very well. This script does indeed fly the camera in a circle around the avatar, as long as `camera_locked=true`. The trouble with that is that the avatar can walk away from the center of the circle, leaving the camera still describing the original circle while watching the avatar walk away.

I had assumed that `camera_locked=false` would mean that the camera would continue to fly in circles around the avatar even as the avatar walks around. Instead it engages in a behavior I find very weird. It hops to some position, comes closer and then moves away, then hops to a different position, comes closer and then moves away, returns to the first position, ...

I'm having trouble explaining this to myself.

Reply:
Unfortunately, I don't have a quick answer, because 'follow cam' is quite new for me too. So I need to dig and test this script myself.

Reply:
> the avatar can walk away from the center of the circle, leaving the camera still describing the original circle while watching the avatar walk away.

Of course there's a simple fix for that, which I've pushed as ["Ensure that the flycam stays near moving avatar."](https://github.com/secondlife/viewer/pull/2209/commits/b4fe47a5c0abac02d161640e04a9a78afb1c5987)